extensions/sql-format/src/index.tsx

```diff
@@ -0,0 +1,179 @@
+import { Clipboard } from '@raycast/api';
+import { getPreferenceValues, showHUD, showToast, Toast } from '@raycast/api';
+
+import { format } from 'sql-formatter';
+
+export default async () => {
+  const output = formatSQL((await Clipboard.readText()) || '');
+  if (output) {
+    await copyFormattedSQL(output);
+  }
+};
+
+// format sql
+export function formatSQL(text: string) {
+  const trimmedText = text.trim();
+  let output = '';
+  try {
+    if (!validateSQL(trimmedText)) {
+      showToastMessage();
+      return;
+    }
+
+    const tabWidth = getTabWidth();
+    const keywordCase = getKeywordCase();
+    const options = {
+      tabWidth: tabWidth === 'tab' ? 8 : parseInt(tabWidth),
+      keywordCase: keywordCase,
+    };
+
+    output = format(trimmedText, options);
+  } catch (error) {
+    showToastMessage();
+    return;
+  }
+  return output;
+}
+
+// copy or auto paste output
+export async function copyFormattedSQL(result: string) {
+  if (autoPasteEnabled()) {
+    await Clipboard.paste(result);
+    await showHUD('✅ Pasted succesfully!');
+  } else {
+    await Clipboard.copy(result);
+    await showHUD('✅ Copied succesfully!');
+  }
```

Comment:
(non-blocking) We can use toasts here in UserInitiated mode for better color coded prompts. See the changes in https://github.com/raycast/extensions/pull/14155 for more details.

Reply:
It is more colorful  after I changed, but I want to know when LaunchType.Background to use it. Beside choose <run in background>, the action is not frequently.
<img width="750" alt="image" src="https://github.com/user-attachments/assets/682a27b5-3441-4106-8ca5-04f5d5c201c7">

Reply:
You're right, in most cases these commands will not run in Background mode. But, to be fault-tolerant in case someone launches the command with [deeplink](https://developers.raycast.com/information/lifecycle/deeplinks) and `launchType=background`, the command will not fail (as Toasts are not available in Background mode).

Reply:
ok, i got it, you explanation is very clear about it.